azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/strategy/CleanupResources.java

```diff
@@ -86,10 +89,10 @@ public boolean cleanupNode(final String id) {
    }
 
    public void cleanupVirtualMachineNICs(VirtualMachine virtualMachine) {
-      for (IdReference nicRef : virtualMachine.properties().networkProfile().networkInterfaces()) {
-         String nicResourceGroup = nicRef.resourceGroup();
-         String nicName = nicRef.name();
-         NetworkInterfaceCard nic = api.getNetworkInterfaceCardApi(nicRef.resourceGroup()).get(nicName);
+      for (NetworkInterface nicRef : virtualMachine.properties().networkProfile().networkInterfaces()) {
```

Comment:
Well, it is actually a reference to a network interface card, since it just holds its ID and a flag indicating if it is the primary one.